aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/gcm/AeroGearGCMPushJsonConfiguration.java

```diff
@@ -0,0 +1,290 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.android.unifiedpush.gcm;
+
+import android.content.Context;
+import android.util.Log;
+import org.jboss.aerogear.android.unifiedpush.PushConfiguration;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A Push Configuration which builds {@link org.jboss.aerogear.android.unifiedpush.gcm.AeroGearGCMPushRegistrar} instances.
+ */
+public class AeroGearGCMPushJsonConfiguration
+        extends PushConfiguration<AeroGearGCMPushJsonConfiguration> {
+
+    private static final String TAG = AeroGearGCMPushJsonConfiguration.class.getName();
+    private final UnifiedPushConfig pushConfig = new UnifiedPushConfig();
+    private String fileName = "push-config.json";
```

Comment:
IMO no, the idea is have a default name but be flexible to use any name you want or in some case can use more than one file to register in more than one app/UPS/whatever